packages/entities/entities-plugins/src/components/PluginSelectCard.vue

```diff
@@ -0,0 +1,310 @@
+<script setup lang="ts">
+import { computed, type PropType } from 'vue'
+import { useRouter } from 'vue-router'
+import { PluginGroup, type KongManagerPluginFormConfig, type KonnectPluginFormConfig, type PluginType } from '../types'
+import { KUI_ICON_SIZE_30, KUI_COLOR_TEXT_NEUTRAL_STRONGER } from '@kong/design-tokens'
+import composables from '../composables'
+import PluginIcon from './PluginIcon.vue'
+
+const emit = defineEmits<{
+  (e: 'plugin-clicked', plugin: PluginType) : void,
+  (e: 'custom-plugin-delete'): void,
+}>()
+
+const props = defineProps({
+  /** The base konnect or kongManger config. Pass additional config props in the shared entity component as needed. */
+  config: {
+    type: Object as PropType<KonnectPluginFormConfig | KongManagerPluginFormConfig>,
+    required: true,
+    validator: (config: KonnectPluginFormConfig | KongManagerPluginFormConfig): boolean => {
+      if (!config || !['konnect', 'kongManager'].includes(config?.app)) return false
+      if (!config.getCreateRoute) return false
+      return true
+    },
+  },
+  /**
+   * Whether or not user has rights to delete custom plugins
+   */
+  canDeleteCustom: {
+    type: Boolean,
+    default: false,
+  },
+  /**
+   * Whether or not user has rights to edit custom plugins
+   */
+  canEditCustom: {
+    type: Boolean,
+    default: false,
+  },
+  plugin: {
+    type: Object as PropType<PluginType>,
+    required: true,
+  },
+  noRouteChange: {
+    type: Boolean,
+    default: false,
+  },
+})
+
+const router = useRouter()
+const { i18n: { t } } = composables.useI18n()
+const controlPlaneId = computed((): string => props.config.app === 'konnect' ? props.config.controlPlaneId : '')
+
+const handleCreateClick = (): void => {
+  router.push(props.config.getCreateRoute(props.plugin.id))
+}
+
+const emitPluginData = (): void => {
+  emit('plugin-clicked', props.plugin)
+}
+
+/**
+ * Custom Plugin logic
+ */
+const isCreateCustomPlugin = computed((): boolean => props.plugin.id === 'custom-plugin-create')
+const isCustomPlugin = computed((): boolean => props.config.app === 'konnect' && props.plugin.group === PluginGroup.CUSTOM_PLUGINS)
+
+const handleCustomDelete = (): void => {
+  if (props.config.app === 'konnect') {
+    emit('custom-plugin-delete')
+  }
+}
+
+const handleCustomEdit = (pluginName: string): void => {
+  if (props.config.app === 'konnect' && props.config.getCustomEditRoute) {
+    router.push(props.config.getCustomEditRoute(pluginName))
+  }
+}
+
+const handleCustomClick = (): void => {
+  // TODO: verify
+  // handle custom plugin card click only
+  if (props.config.app === 'konnect') {
+    if (isCreateCustomPlugin.value && props.config.createCustomRoute) {
+      router.push(props.config.createCustomRoute)
+    } else if (isCustomPlugin.value) {
+      handleCreateClick()
+    }
+  }
+}
+</script>
+
+<style lang="scss" scoped>
+.plugin-card {
+  color: initial;
+  display: flex;
+  flex-basis: 100%;
+  flex-flow: row-wrap;
+  max-width: 335px;
```

Comment:
question: why this specific max-width?

Reply:
I'm assuming we don't want the cards themselves to be resizing based on the space available. This is the width that seems to look best for the design?

![image](https://github.com/Kong/public-ui-components/assets/67973710/db09baeb-dbe2-4e41-97a6-4df0d71750f4)

Reply:
Could we use CSS grid to handle the size instead of flex?

Reply:
I don't have time to refactor the CSS right now and this isn't something I know how to do off the top of my head. Could it be a ticket for the backlog?